src/shogun/base/SGObject.h

```diff
@@ -448,6 +450,9 @@ class CSGObject
 	void put(const std::string& name, T* value)
 	{
 		put(Tag<T*>(name), value);
+		std::string callback_fun = name + "_callback";
+		if(has(callback_fun))
```

Comment:
one thing I am skeptical about is that this is a bit intrusive for the existing ::put methods.
Also, would there be another thing using these callbacks? Because if not this seems like a lot of architecture for a simple thing like recursively applying seeds.

I have the feeling maybe we should add an `init_seed()` method in SGObject, which does the recursive seed setting (if the parameter exists). Then everything would be in one place at least and it would be a NOP for non-random classes (could even warn/error in that case). This would not change "put" at all, but on the other hand would pollute CSGObject. Although that could be solved with a function to init the seed, e.g. `init_seed(my_object)`, which would behave as described above.

There are obviously pro/con arguments for each approach

Reply:
I thought this is less ad-hoc, since we could benefit from callbacks in other classes, and also to unify the way parameters are set. not sure though